### DISCUSSION TITLE
OpenAMP comm between linux processes

### DISCUSSION THREAD

**Discussion by gsnayak90** [OpenAMP/open-amp#473] [Q&A]
Hi,
I am trying to get a simple echo demo app running on linux ( both client and server). following steps are followed -  https://github.com/OpenAMP/open-amp#example-to-compile-openamp-for-communication-between-linux-processes

I get the following error when test server is started ( rpmsg-echo-shared app ) on linux
$ sudo LD_LIBRARY_PATH=<openamp_built>/usr/local/lib:<libmetal_built>/usr/local/lib \
   build/usr/local/bin/rpmsg-echo-shared

openamp lib version: 1.3.0 (Major: 1, Minor: 3, Patch: 0)
libmetal lib version: 1.3.0 (Major: 1, Minor: 3, Patch: 0)
Starting application...
metal: debug:     added page size 4096 @/tmp
Error in metal_init
metal: error:     failed pagemap pread (offset 3fdb996880) - Illegal seek
metal: error:     failed pagemap pread (offset 3fdb996888) - Illegal seek
metal: error:     failed pagemap pread (offset 3fdb996890) - Illegal seek
...
metal: error:     failed pagemap pread (offset 3fa0381ce0) - Illegal seek
metal: error:     failed pagemap pread (offset 3fa0381ce8) - Illegal seek
metal: error:     failed pagemap pread (offset 3fa0381cf0) - Illegal seek
metal: error:     failed pagemap pread (offset 3fa0381cf8) - Illegal seek
Waiting for connection on /tmp/openamp.event.0

Am I missing any steps here?
Thanks,

**Comment by arnopo** [comment]
you have to start 2 processes: one to echo, one to send messages

```
LD_LIBRARY_PATH=<openamp_built>/usr/local/lib:<libmetal_built>/usr/local/lib  ./bin/usr/local/bin/rpmsg-echo-shared >/dev/null&
sleep 1
LD_LIBRARY_PATH=<openamp_built>/usr/local/lib:<libmetal_built>/usr/local/lib  ./bin/usr/local/bin/rpmsg-echo-ping-shared 1 || exit
```

**Reply by gsnayak90** [reply]
Hi @arnopo 
Thanks for your reply..

When I do "LD_LIBRARY_PATH=<openamp_built>/usr/local/lib:<libmetal_built>/usr/local/lib  ./bin/usr/local/bin/rpmsg-echo-shared ", i get the following error  ( without having to run the second app to send a message). Added couple of prints in metal_init() and it seems to be failing - 

openamp lib version: 1.3.0 (Major: 1, Minor: 3, Patch: 0)
libmetal lib version: 1.3.0 (Major: 1, Minor: 3, Patch: 0)
Starting application...
metal: debug:     added page size 4096 @/tmp
Error in metal_init
metal: error:     failed pagemap pread (offset 3f8e599880) - Illegal seek
metal: error:     failed pagemap pread (offset 3f8e599888) - Illegal seek
metal: error:     failed pagemap pread (offset 3f8e599890) - Illegal seek
metal: error:     failed pagemap pread (offset 3f8e599898) - Illegal seek
metal: error:     failed pagemap pread (offset 3f8e5998a0) - Illegal seek
metal: error:     failed pagemap pread (offset 3f8e5998a8) - Illegal seek
metal: error:     failed pagemap pread (offset 3f8e5998b0) - Illegal seek
metal: error:     failed pagemap pread (offset 3f8e5998b8) - Illegal seek
metal: error:     failed pagemap pread (offset 3f8e5998c0) - Illegal seek
....

Looking at other threads, found this one - https://github.com/OpenAMP/open-amp/discussions/456#discussioncomment-4330728
Is it in anyway related?

Thanks,

**Comment by arnopo** [comment]
Could you detail your environment? are you run the sample on a PC or on a board,  which distribution, ...?

**Reply by gsnayak90** [reply]
Its on a PC with ubuntu over wsl 
ubuntu version - Ubuntu 22.04.1 LTS

**Reply by arnopo** [reply]
No such issue on my PC native Linux PC running Ubuntu 20.04.
is the issue linked to WSL? 
The logs correspond to this function:
https://github.com/OpenAMP/libmetal/blob/main/lib/system/linux/utilities.c#L263

Seems to me that the offset computed is not valid. 
The issue seems around the shmem allocation based on file system for the linux, that needs some adaptations for your system.
